src/structures/Interaction.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+const Base = require('./Base');
+const { InteractionTypes } = require('../util/Constants');
+const SnowflakeUtil = require('../util/SnowflakeUtil');
+
+/**
+ * Represents an interaction.
+ * @extends {Base}
+ */
+class Interaction extends Base {
+  constructor(client, data) {
+    super(client);
+
+    /**
+     * The type of this interaction
+     * @type {InteractionType}
+     */
+    this.type = InteractionTypes[data.type];
+
+    /**
+     * The ID of this interaction
+     * @type {Snowflake}
+     */
+    this.id = data.id;
+
+    /**
+     * The token of this interaction
+     * @type {string}
+     * @name Interaction#token
+     * @readonly
+     */
+    Object.defineProperty(this, 'token', { value: data.token });
+
+    /**
+     * The ID of the channel this interaction was sent in
+     * @type {?Snowflake}
+     */
+    this.channelID = data.channel_id ?? null;
+
+    /**
+     * The ID of the guild this interaction was sent in
+     * @type {?Snowflake}
+     */
+    this.guildID = data.guild_id ?? null;
+
+    /**
+     * The user which sent this interaction
+     * @type {User}
+     */
+    this.user = this.client.users.add(data.user ?? data.member.user);
+
+    /**
+     * If this interaction was sent in a guild, the member which sent it
+     * @type {?GuildMember}
+     */
+    this.member = data.member ? this.guild?.members.add(data.member) ?? null : null;
```

Comment:
Why not support raw data here?

Reply:
I'm personally not a fan of forwarding raw objects to the user. It would help if the guild member and message classes both just had a `guildID` property and a `guild` getter but i suppose that's not acievable before -next, so using the raw member instead is probably the better idea.

Reply:
Why is it not doable before -next? it's a new thing, surely we can do it 👀 Plus, we already do raw data for webhooks

Reply:
I'll add support for raw data as soon as #5539 gets merged